### PR TITLE
06065 - renamed `stake` to `weight` throughout the platform

### LINKED ISSUE #6065: change updateStake to updateWeight
In order to disambiguate the concept of consensus weight from per node from the concept of stake per node, the fields and methods that reference `stake` in the address book should be changed to `weight`.   
* SwirldState.updateStake()
* AddressBook.updateStake()
* AddressBook.getTotalStake()
* Address.getStake()
* Address.copySetStake()

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:
All references to `stake` are changed to `weight` at the platform level.  This is to disambiguate the concept of hbar staked to nodes and the weight each node has in contributing to consensus.   The two concepts are related, but they are different and using the same word can be confusing.   

**Related issue(s)**:

Fixes #6065 

**Notes for reviewer**:
The platform builds and can run the demo app.  

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/emergency/EmergencySignedStateValidator.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 /**
  * Validates the signed state received by the learner in an emergency reconnect. If the received state has the exact
  * round and hash requested, it does not need to be fully signed. If the state is for a later round, it must be signed
- * by at least half the network stake to be considered valid. The emergency reconnect scenario is described in
+ * by at least half the network Weight to be considered valid. The emergency reconnect scenario is described in
```
Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/emergency/EmergencySignedStateValidator.java:36]:
```suggestion
 * by at least half the network weight to be considered valid. The emergency reconnect scenario is described in
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/emergency/EmergencySignedStateValidator.java [hunk 2]
```diff
@@ -133,7 +133,7 @@ private void verifyLaterRoundIsValid(final SignedState signedState, final Addres
         logger.info(
                 SIGNED_STATE.getMarker(),
                 "Emergency recovery signed state is for round later than requested. "
-                        + "Validating that the state is signed by a majority stake.");
+                        + "Validating that the state is signed by a majority Weight.");
```
Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/reconnect/emergency/EmergencySignedStateValidator.java:136]:
```suggestion
                        + "Validating that the state is signed by a majority weight.");
```

File: platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/AddressBookInitializerTest.java [hunk 3]
```diff
@@ -187,11 +187,11 @@ void currentVersionEqualsStateVersion() throws IOException {
     }
 
     @Test
-    @DisplayName("Version upgrade, SwirldState set 0 Stake.")
-    void versionUpgradeSwirldStateZeroStake() throws IOException {
+    @DisplayName("Version upgrade, SwirldState set 0 Weight.")
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/AddressBookInitializerTest.java:190]:
```suggestion
    @DisplayName("Version upgrade, SwirldState set 0 weight.")
```

File: platform-sdk/sdk/config.txt [hunk 4]
```diff
@@ -128,7 +128,7 @@ swirld, 123
 # (after the word "address" at the start):
 #         nickname            - a short version of the member's name
 #         full name           - the member's name
-#         stake               - a nonnegative integer that weighting the consensus voting
+#         weight              - a nonnegative integer that weighting the consensus voting
```
Comment (jsync-swirlds) [inline platform-sdk/sdk/config.txt:131]: weighting ==> weights

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java [hunk 5]
```diff
@@ -148,7 +148,7 @@ private byte[] clone(byte[] x) {
      * @param id                  the ID for that member
      * @param nickname            the name given to that member by the member creating this address
      * @param selfName            the name given to that member by themself
-     * @param stake               the amount of stake (0 if they should have no influence on the consensus)
+     * @param weight               the amount of weight (0 if they should have no influence on the consensus)
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java:151]: spacing off.
`* @param weight               the amount of weight (0 if they should have no influence on the consensus)`
should be
`* @param weight              the amount of weight (0 if they should have no influence on the consensus)`

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java [hunk 6]
```diff
@@ -209,7 +209,7 @@ public Address(
...
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java:212]: extra space (same as 151).

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java [hunk 7]
```diff
@@ -136,7 +136,7 @@ public AddressBook getInitialAddressBook() {
 
     /**
      * Determines the address book to use.  If the configured current version of the application is higher than the save
-     * state version, the swirld state is given the config address book to determine stake weights.  If the address book
+     * state version, the swirld state is given the config address book to determine weight weights.  If the address book
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java:139]: `weight weights` doubles the object. Should be `weights`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/internal/ConsensusHashFinder.java [hunk 8]
```diff
@@ -205,28 +205,28 @@ public Map<Hash, HashPartition> getPartitionMap() {
...
      */
     public void writePartitionData(final StringBuilder sb) {
         sb.append("Nodes holding ")
-                .append(hashReportedStake)
+                .append(hashReportedWeight)
                 .append("/")
-                .append(totalStake)
-                .append(" stake have reported a hash for round ")
+                .append(totalWeight)
+                .append(" weight have reported a hash for round ")
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/internal/ConsensusHashFinder.java:229]: A bit awkward wording, would read better as `weight of nodes have reported a hash for round`

File: platform-sdk/swirlds-platform-core/src/test/resources/com/swirlds/platform/config/legacy/config.txt [hunk 9]
```diff
@@ -131,7 +131,7 @@ swirld, 123
 # (after the word "address" at the start):
 #         nickname            - a short version of the member's name
 #         full name           - the member's name
-#         stake               - a nonnegative integer that weighting the consensus voting
+#         weight              - a nonnegative integer that weighting the consensus voting
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-platform-core/src/test/resources/com/swirlds/platform/config/legacy/config.txt:134]: `that weighting` ==> `weighting`

**Review by jsync-swirlds — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/internal/ConsensusHashFinder.java (hunk 8 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/internal/ConsensusHashFinder.java:229]: the phrases is embedded in a larger string builder.  This is the total statement: 
``` 
Nodes holding (# partial)/(# total) weight have reported a hash for round (# round).
```
vs 
```
Nodes holding (# partial)/(# total) weight of nodes have reported a hash for round (# round).
```
I am deciding to go with the following: 
``` 
Nodes holding (#)/(#) of total weight have reported a hash for round (#).
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java (hunk 7 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java:139]: deleting the first weight

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java (hunk 6 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java:212]: removing the space.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java (hunk 5 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java:151]: removing the space.

**Review by jsync-swirlds — DISMISSED** [review]

**Review by alittley — DISMISSED** [review]

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolMain.java [hunk 10]
```diff
@@ -34,7 +34,7 @@
 
 /**
  * <p>
- * An application that updates address book stake weights on version upgrade.
+ * An application that updates address book weight weights on version upgrade.
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolMain.java:37]: removed `weight` in next commit.

**Review by alittley — APPROVED** [review]

File: platform-sdk/sdk/config.txt (hunk 4 above)
Comment (edward-swirldslabs) [inline platform-sdk/sdk/config.txt:131]:
```suggestion
#         weight              - a nonnegative integer that weights the consensus voting
```

**Review by alittley — APPROVED** [review]

**Pull request merged**